[SPIRV] Do not remove capability from all caps

pmatos opened the pull request:
We were removing bit_instructions cap from All caps but this was a mistake.

Test SPV_KHR_bit_instructions was failing. Remove function removeCapabilityIf. It was not being done correctly and is now unnecessary.

iliya-diyachkov commented:
Why do you think, it is a mistake? Well, SPIRV translator emits both capabilities in the original test. However I built the original test without `OpCapability Shader` in SPIRV asm and SPIRV validator didn't report errors...

pmatos commented:
> Why do you think, it is a mistake? Well, SPIRV translator emits both capabilities in the original test. However, I built the test without `OpCapability Shader` in SPIRV asm, and SPIRV validator didn't report errors... If I remove the both capabilities, the validator reports the error:
> 
> ```
> error: line 34: Opcode BitReverse requires one of these capabilities: Shader BitInstructions 
>   %call = OpBitReverse %uint %b
> ```

I initially added this code but I think I misinterpreted the purpuse of `AvailableCaps`. We are _not_ removing both capabilities. We are just not removing BitInstructions. We make sure they are both _available_. 

Now all relevant tests pass. If we use bit instructions but no instructions that require shader we use the bit instructions extension. If we use instructions that require shader as well, we use the shader extension but not bit instructions. We just make sure they are exclusive to each other. The test `llvm/test/CodeGen/SPIRV/extensions/SPV_KHR_bit_instructions.ll` should be self-explanatory.

iliya-diyachkov commented:
> I initially added this code but I think I misinterpreted the purpuse of AvailableCaps.
Well, we had to use `AllCaps` instead of `AvailableCaps`.

pmatos commented:
> > I initially added this code but I think I misinterpreted the purpuse of AvailableCaps.
> 
> Well, we needed to use `AllCaps` instead of `AvailableCaps`.

You are correct yes, I should have used AllCaps instead of AvailableCaps.

pmatos committed:
[SPIRV] Fix test SPV_KHR_bit_instructions

Use AllCAps instead of AvailableCaps as initial implementation
confused these two.
Also document their purpose and difference.

pmatos commented:
@iliya-diyachkov much simpler now, wdyt?

Commit:
Run clang-format

iliya-diyachkov commented:
Sorry for delay with the answer, Paulo.
Yes, it looks fine.

pmatos commented:
> Sorry for delay with the answer, Paulo. Yes, it looks fine.

Thanks.

iliya-diyachkov reviewed (approved):
It looks good to me. If the build and testing are passed, you can merge the patch.
